GCRCatalogs/user_config_mgr.py

```diff
@@ -0,0 +1,123 @@
+import os
+import warnings
+import yaml
+from collections.abc import MutableMapping
+
+"""
+Utility class for managing user config (a dict) persisted to a yaml file. 
+"""
+
+__all__ = ['UserConfigManager']
+
+
+class UserConfigManager(MutableMapping):
+    _default_config_filename = "gcr-catalogs.yaml"
+    _default_config_reldir = "lsstdesc"
+    
+    def __init__(self, config_filename=None, config_reldir=None):
+        config_name = os.path.basename(config_filename or self._default_config_filename)
+        reldir = config_reldir or self._default_config_reldir
+        if os.path.isabs(reldir):
+            raise ValueError(f'{reldir} must be a relative path')
+        self._user_config_dir = os.path.join(self._get_config_home(), reldir)
+        self._user_config_path = os.path.join(self._user_config_dir, config_name)
+
+    @staticmethod
+    def _get_config_home():
+        if os.getenv("XDG_CONFIG_HOME"):                   # Unix
+            return(os.getenv("XDG_CONFIG_HOME"))	
```

Comment:
Removing trailing tabs
```suggestion
            return(os.getenv("XDG_CONFIG_HOME"))
```